middlewares/redirect_as2.ts

```diff
@@ -0,0 +1,68 @@
+import type { Middleware } from "../core/server.ts";
+
+export type Options = (url: URL) => Promise<URL> | URL | undefined | void;
```

Comment:
Options should be an object for consistency with all other Lume's plugins and middlewares. It allows to extend it with extra options in the future.